core: thread_state_t makes RIOT not compilable on MAC OSX

<!--
====================================
IF YOUR ISSUE IS RELATED TO SECURITY
====================================
please submit it to the security mailing-list security@riot-os.org.

If your issue is a question related to the usage of RIOT, please submit it to
the user mailing-list users@riot-os.org or to the developer mailing-list
devel@riot-os.org.
-->

#### Description
<!--
Example: Cannot build gnrc_networking application for samr21-xpro board.
-->

Since #10989 was merged, most RIOT applications don't compile on MAC anymore, as there is a type `thread_state_t` also defined in [Apples thread handler headers](https://opensource.apple.com/source/xnu/xnu-792/osfmk/mach/thread_status.h.auto.html). As a fix I propose to rename that type to something that collides less with that particular type.

#### Steps to reproduce the issue
Build e.g. `examples/gnrc_networking` for `native` on a MAC-OSX-based machine.
<!--
Try to describe as precisely as possible here the steps required to reproduce
the issue. Here you can also describe your hardware configuration, the network
setup, etc.
-->

#### Expected results
The application should compile.
<!--
Example: The gnrc_networking application builds on samr21-xpro.
-->

#### Actual results
The application can't compile with the following error:

```
/Users/mlenders/RIOT/core/include/sched.h:117:3: error: typedef redefinition with different types ('enum thread_state_t' vs 'natural_t *' (aka 'unsigned int *'))
} thread_state_t;
  ^
/usr/include/mach/thread_status.h:83:20: note: previous definition is here
typedef natural_t       *thread_state_t;        /* Variable-length array */
                         ^
1 error generated.
```
<!--
Please paste or specifically describe the actual output.
-->

#### Versions
```
Operating System Environment
-----------------------------
       Operating System: Mac OS X 10.13.6
                 Kernel: Darwin 17.7.0 x86_64 i386

Installed compiler toolchains
-----------------------------
             native gcc: Apple LLVM version 10.0.0 (clang-1000.10.44.4)
      arm-none-eabi-gcc: missing
                avr-gcc: missing
       mips-mti-elf-gcc: missing
             msp430-gcc: missing
   riscv-none-embed-gcc: missing
   xtensa-esp32-elf-gcc: missing
   xtensa-lx106-elf-gcc: missing
                  clang: Apple LLVM version 10.0.0 (clang-1000.10.44.4)

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: missing
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
xtensa-esp32-elf-newlib: missing
xtensa-lx106-elf-newlib: missing
               avr-libc: missing (missing)

Installed development tools
---------------------------
                  cmake: missing
               cppcheck: missing
                doxygen: 1.8.4
                 flake8: missing
                    git: git version 2.17.2 (Apple Git-113)
                   make: GNU Make 3.81
                openocd: missing
                 python: Python 2.7.10
                python2: missing
                python3: Python 3.7.3
             coccinelle: missing
```
<!--
Operating system: Mac OSX, Linux, Vagrant VM
Build environment: GCC, CLang versions (you can run the following command from
the RIOT base directory: ./dist/tools/ci/print_toolchain_versions.sh).
-->

<!-- Thanks for contributing! -->

--- a/core/include/sched.h
+++ b/core/include/sched.h
@@ -114,7 +114,7 @@ typedef enum {
     STATUS_RUNNING,                 /**< currently running                    */
     STATUS_PENDING,                 /**< waiting to be scheduled to run       */
     STATUS_NUMOF                    /**< number of supported thread states    */
-} thread_state_t;
+} thread_status_t;
 /** @} */
 
 /**
@@ -123,7 +123,7 @@ typedef enum {
  */
 #define STATUS_ON_RUNQUEUE      STATUS_RUNNING  /**< to check if on run queue:
                                                  `st >= STATUS_ON_RUNQUEUE`   */
-#define STATUS_NOT_FOUND ((thread_state_t)-1)   /**< Describes an illegal thread status */
+#define STATUS_NOT_FOUND ((thread_status_t)-1)  /**< Describes an illegal thread status */
 /** @} */
 /**
  * @def SCHED_PRIO_LEVELS
@@ -146,7 +146,7 @@ int sched_run(void);
  *                          targeted process
  * @param[in]   status      The new status of this thread
  */
-void sched_set_status(thread_t *process, thread_state_t status);
+void sched_set_status(thread_t *process, thread_status_t status);
 
 /**
  * @brief       Yield if approriate.

--- a/core/include/thread.h
+++ b/core/include/thread.h
@@ -143,7 +143,7 @@ typedef void *(*thread_task_func_t)(void *arg);
  */
 struct _thread {
     char *sp;                       /**< thread's stack pointer         */
-    thread_state_t status;          /**< thread's status                */
+    thread_status_t status;         /**< thread's status                */
     uint8_t priority;               /**< thread's priority              */
 
     kernel_pid_t pid;               /**< thread's process id            */

--- a/core/sched.c
+++ b/core/sched.c
@@ -158,7 +158,7 @@ void sched_register_cb(void (*callback)(uint32_t, uint32_t))
 }
 #endif
 
-void sched_set_status(thread_t *process, thread_state_t status)
+void sched_set_status(thread_t *process, thread_status_t status)
 {
     if (status >= STATUS_ON_RUNQUEUE) {
         if (!(process->status >= STATUS_ON_RUNQUEUE)) {

--- a/sys/ps/ps.c
+++ b/sys/ps/ps.c
@@ -96,7 +96,7 @@ void ps(void)
         thread_t *p = (thread_t *)sched_threads[i];
 
         if (p != NULL) {
-            thread_state_t state = p->status;                                      /* copy state */
+            thread_status_t state = p->status;                                     /* copy state */
             const char *sname = state_names[state];                                /* get state name */
             const char *queued = &queued_name[(int)(state >= STATUS_ON_RUNQUEUE)]; /* get queued flag */
 #ifdef DEVELHELP

